client/packages/system/src/Item/context/useUnitVariant.ts

```diff
@@ -75,10 +91,11 @@ export const useUnitVariant = (
     ],
     isEqual
   );
+  const t = useTranslation('common');
 
   if (!item || item.variants.length == 0) {
     return {
-      asPackUnit: packSize => String(packSize),
+      asPackUnit: packSize => asPackUnit({ packSize, unitName, t }),
```

Comment:
I found it hard to read the code, having two variations of the method `asPackUnit` with different props. Perhaps rename one?

Also - the TypedTFunction is passed in to the method, only to translate `'unit.name'`. 
I'd create the function within the component, so that it has access to the translate function, or pass in the translated string

Reply:
> I found it hard to read the code, having two variations of the method asPackUnit with different props. Perhaps rename one?

Agree, also was concerned when writing it: cd1739ee2082150406a0e89c0c34fc8b2d345c7a

> Also - the TypedTFunction is passed in to the method, only to translate 'unit.name'.
I'd create the function within the component, so that it has access to the translate function, or pass in the translated string

I kind of like the encapsulation of entire logic in one place, rather then needing to look up method references to figure out what the default unit should be, also makes the consumption easier. Will merge as is, let me know if you feel strong about this.